BHoM_UI/Caller/IO_CRUD.cs

```diff
@@ -75,20 +75,25 @@ public virtual bool RemoveInput(string name)
 
         /*************************************/
 
-        public virtual bool UpdateInput(int index, string name, Type type = null)
+        public virtual bool UpdateInput(int index, string name = null, Type type = null)
         {
+            if (index < 0)
+                return false;
+
             if (InputParams.Count <= index)
                 return AddInput(index, name, type);
 
-            if (type != InputParams[index].DataType)
-                m_CompiledGetters[index] = Engine.UI.Create.InputAccessor(m_DataAccessor.GetType(), type);
-
             if (name != null)
                 InputParams[index].Name = name;
 
             if (type != null)
+            {
                 InputParams[index].DataType = type;
 
+                if (type != InputParams[index].DataType)
```

Comment:
code message was not posted for whatever reason:

This if should always be true given the parameter is assigned just above.

Not sure if the execution order is important, as in, a new compilation should only happen if the input has actually changed, then I would move this above line 90. If not, I think we should jsut delete the if.

Reply:
Well, that happened :disappointed:

Now fixed